Assets/Dependencies/SmashBrew/Character/States/MovementState.cs

```diff
@@ -168,16 +168,18 @@ struct MovementInfo {
 
         void SnapToLedge() {
             IsFastFalling = false;
+            var ledge = CurrentLedge.GetComponentInParent<StageLedge>();
```

Comment:
Oh, the problem is that it's called every frame. I could just move this code out into the setter for CurrentLedge instead. I just thought it would be more logical for the direction to be set during the snapping method.